controllers/ccruntime_controller.go

```diff
@@ -57,13 +57,27 @@ type CcRuntimeReconciler struct {
 	Namespace string
 }
 
+const (
+	peerpodConfigCRName = "peerpodconfig-coco"
+	DEFAULT_PEER_PODS   = "10"
+)
+
 //+kubebuilder:rbac:groups=confidentialcontainers.org,resources=ccruntimes,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=confidentialcontainers.org,resources=ccruntimes/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=confidentialcontainers.org,resources=ccruntimes/finalizers,verbs=update
 //+kubebuilder:rbac:groups=core,resources=nodes,verbs=get;list;watch;patch;update
 //+kubebuilder:rbac:groups=apps,resources=daemonsets,verbs=get;list;watch;create;delete;update;patch
 //+kubebuilder:rbac:groups=node.k8s.io,resources=runtimeclasses,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;update
+//+kubebuilder:rbac:groups="",resources=nodes/status,verbs=patch
+//+kubebuilder:rbac:groups=confidentialcontainers.org,resources=peerpodconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=confidentialcontainers.org,resources=peerpodconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=confidentialcontainers.org,resources=peerpodconfigs/finalizers,verbs=update
+//+kubebuilder:rbac:groups=confidentialcontainers.org,resources=peerpods,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=confidentialcontainers.org,resources=peerpods/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=confidentialcontainers.org,resources=peerpods/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=create;get;update;list;watch
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=create;get;update;list;watch
```

Comment:
these tags do not belong here.

Reply:
@mythi 
OK you mean that I should get rid of the peerpods subdirectory and combine all the peerpod config yamls into one patch.yaml. So config/default/kustomization.yaml would contain:

# [PEERPODS] To enable the peerpod controllers uncomment this section.
#- manager_peerpods_patch.yaml

As far as the rbac permissions goes you want those moved from the go file to this manager_peerpods_patch.yaml as well. Could you show me an example as I am not familiar with implementing rbac in this way.

Reply:
I was thinking something like `config/overlays/peerpods/kustomization.yaml` with

```
resources:
- ../../default
- <rbac from cloud-api-adator>
- <peerpods crds from cloud-api-adaptor>
- any_other_yaml_needed_here.yaml
```

Reply:
Whats to prevent it from being pulled in when peerpods is not selected?

Reply:
And you want me to move the crds out of config/crd/kustomization.yaml which I just changed to an upstream pull?

Reply:
> Whats to prevent this new yaml from being pulled in when peerpods is not selected?

`kubectl -k config/overlays/peerpods` would not be used by those who don't want/need peerpods. They continue to use `config/default` without even noticing peerpods was added.

> And you want me to move the crds out of config/crd/kustomization.yaml which I recently changed to an upstream pull as well?

Yes, that's not a good place for those. They can be part of the peerpods overlay. My thinking is like this: peerpods APIs have their own lifecycle in that other repository. The APIs (CRDs), the corresponding controllers, necessary RBAC rules etc. are maintained elsewhere and this repo just consumes releases from peerpods repo. The same way `go.mod` dependencies are updated for the controller code, the corresponding yamls are pulled via tagged kustomize rules.

Reply:
OK I'll give it a whirl.

Reply:
@mythi @jensfr 
I've been able to separate out most things to an overlays/peerpod directory with 3 subdirectories ( crd, rbac, default ). One reason I created separate directories is the crds are applied at install and not deploy with the other yamls. I am having difficulty with the rbac separation. Specifically pulling from the upstream confidential-containers/cloud-api-adaptor/peerpodconfig-ctrl and confidential-containers/cloud-api-adaptor/peerpod-ctrl rbac directories. These files have openshift stuff in them and are not generic. What I would like to do is go ahead and place all the rbac files I need for the peerpod controllers in the rbac overlays subdirectory and create a CAA Issues request for a generic isolation of the peerpod controller's rbacs.

Reply:
@caavery did you push the updates for review?

Reply:
@mythi @jensfr 
I have pushed the patches addressing the latest review comments with the exception of importing the peerpod rbacs as proposed in https://github.com/confidential-containers/operator/pull/249#discussion_r1365754689

Reply:
The latest version looks good to me, thanks @caavery!

Reply:
@mythi 

Thank you very much for all your help! BTW the PR doesn't seem to be registering your approval. 

Take Care,
Cathy